internal/controller/controllers/agent_controller.go

```diff
@@ -174,14 +167,10 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 			// Retrieve cluster by ClusterDeploymentName from the database
 			cluster, err2 := r.Installer.GetClusterByKubeKey(kubeKey)
 			if err2 != nil {
-				if errors.Is(err2, gorm.ErrRecordNotFound) {
-					// Delete the agent, using a finalizer with pre-delete to deregister the host.
-					log.Infof("Cluster name: %s namespace: %s not found in backend, deleting Agent",
-						agent.Spec.ClusterDeploymentName.Name, agent.Spec.ClusterDeploymentName.Namespace)
-					return r.deleteAgent(ctx, log, req.NamespacedName)
-				}
+				log.WithError(err2).Errorf("Fail to get cluster name: %s namespace: %s in backend",
```

Comment:
just wondering: backend is doing all the same validations before binding: checks that cluster exists, checks if it is in appropriate state, etc. What if we just call BindHost in case h.ClusterID == nil , and handle the return error as we handle it right now?

Reply:
I guess, it could work. You want me to change it?

Reply:
yes, let's do it.  no need to do the same validation twice

Reply:
Second thoughts on it. I don't have the Cluster ID so I need to retrieve the cluster by kube key first.

Reply:
you are right, scratch that